### DISCUSSION TITLE
Branches added to the behavior tree at runtime do not show up in debugger

### DISCUSSION THREAD

**Discussion by hunttis** [bitbrain/beehave#200] [General]
I'm not sure if I'm doing something naughty with this plugin, but in my game I have player team members who are either leaders or followers. The leaders do the exploring and the followers follow the leader. This makes their movement a bit different, so I decided I'd make the movement type a separate scene (leaderMovement / followerMovement), which I add into the behavior tree (movementTree) based on the role of the character. This is to avoid creating huge trees where parts of the tree are never gone into depending on the role of the character.

It works really nicely apart from the branches not showing up in the debugger.

Am I doing it wrong?

**Comment by bitbrain** [comment]
Beehave has been built in mind with dynamic adding/removing nodes to the behaviuor tree. What you are describing sounds like a bug to me - could you raise the issue here? https://github.com/bitbrain/beehave/issues/new?assignees=&labels=%F0%9F%90%9B+bug&projects=&template=bug_report.md&title=
